plugins/item_tasks/server/rest.py

```diff
@@ -168,6 +168,30 @@ def _transformOutputs(self, outputs, token, job):
 
         return transformed
 
+    def _validateOutputParentType(self, outputId, parentType, outputSpec):
+        """
+        Checks if the output parent type is compatible with the output type.
+        """
+
+        # Find the corresponding output specification for the given outputID
+        output = None
+        for out in outputSpec:
+            if outputId == out['id']:
+                output = out
+                break
+
+        # If a corresponding output is found, check if its parent type is valid
+        if output:
```

Comment:
I actually really like the for/else language feature, but agree that it is confusing because it's unfamiliar since it's not present in other programming languages. That said, it's a language construct that exists precisely for these sorts of cases, and it's decidedly more elegant than the more C-like construct that is currently written.

I agree with the article that the main source of confusion is the term `else` as opposed to a more explicit one like `nobreak`.

TLDR: While I personally would have written this using for/else, I do not feel that it needs to be changed since it works as written since there are tradeoffs.